app/assets/javascripts/3dbio_viewer/src/data/repositories/EbiDbModelRepository.ts

```diff
@@ -47,6 +54,10 @@ export class EbiDbModelRepository implements DbModelRepository {
                 .value()
         );
     }
+    upload(options: UploadOptions): Boolean {
+        const uploadResults = request(config.upload.searchUrl, options);
+        return true;
```

Comment:
It's important to return a Future here. A future is an async task similar to a Promise, but with typed errors. Promise<Data>, but Future<Error, Data>. Also, they are cancellable, typically used as with a `return` as the last step of a `React.useEffect`. Another important point: a Future is not executed until future.run(onSucces, onError) is called (tipically, the use case returns a future, and the component calls run).